File: frontend/src/api/risk.ts
import api from '../lib/axios'
import type { ApiResponse } from '../types'

export interface RiskAssessment {
  id: string
  entityType: string
  entityId: string
  riskLevel: string
  riskScore: number
  riskFactors: Record<string, boolean | number | string>
  assessedBy?: string
  notes?: string
  assessedAt: string
  createdAt: string
}

export interface ManualOverrideDto {
  risk_level: 'low' | 'medium' | 'high' | 'critical'
  notes?: string
}

export const riskApi = {
  getKYCRisk: (id: string) =>
    api.get<ApiResponse<RiskAssessment>>(`/kyc/${id}/risk`),

  getKYBRisk: (id: string) =>
    api.get<ApiResponse<RiskAssessment>>(`/kyb/${id}/risk`),

  overrideKYCRisk: (id: string, data: ManualOverrideDto) =>
    api.post<ApiResponse<RiskAssessment>>(`/kyc/${id}/risk`, data),

  overrideKYBRisk: (id: string, data: ManualOverrideDto) =>
    api.post<ApiResponse<RiskAssessment>>(`/kyb/${id}/risk`, data),

  listKYCHistory: (id: string) =>
    api.get<ApiResponse<RiskAssessment[]>>(`/kyc/${id}/risk/history`),

  listKYBHistory: (id: string) =>
    api.get<ApiResponse<RiskAssessment[]>>(`/kyb/${id}/risk/history`),
}

File: backend/internal/domain/risk.go
package domain

import (
	"context"
	"time"

	"github.com/google/uuid"
)

type RiskLevel string

const (
	RiskLevelLow      RiskLevel = "low"
	RiskLevelMedium   RiskLevel = "medium"
	RiskLevelHigh     RiskLevel = "high"
	RiskLevelCritical RiskLevel = "critical"
)

type RiskAssessment struct {
	ID          uuid.UUID
	EntityType  string
	EntityID    uuid.UUID
	RiskLevel   RiskLevel
	RiskScore   int
	RiskFactors map[string]any
	AssessedBy  *uuid.UUID
	Notes       string
	AssessedAt  time.Time
	CreatedAt   time.Time
	UpdatedAt   time.Time
}

type RiskRepository interface {
	Create(ctx context.Context, r *RiskAssessment) error
	FindLatestByEntity(ctx context.Context, entityType string, entityID uuid.UUID) (*RiskAssessment, error)
	ListByEntity(ctx context.Context, entityType string, entityID uuid.UUID) ([]RiskAssessment, error)
	UpdateKYCRisk(ctx context.Context, kycID uuid.UUID, level RiskLevel, score int) error
	UpdateKYBRisk(ctx context.Context, kybID uuid.UUID, level RiskLevel, score int) error
}
